### PR TITLE
Add e2e tests for RHEL 9 EFI on Baremetal

### PR DESCRIPTION
**Pull request by 2ez4szliu**
*Issue #, if available:

*Description of changes:*
- Add simple workflow test cases for RHEL 9 EFI on Baremetal.
- Documentation improvement for RHEL 9 on Baremetal.

*Testing (if applicable):*

*Documentation added/planned (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/getting-started/baremetal/bare-spec.md [hunk 1]
```diff
@@ -306,7 +306,7 @@ In the example, there are `TinkerbellMachineConfig` sections for control plane (
 The following fields identify information needed to configure the nodes in each of those groups.
 >**_NOTE:_** Currently, you can only have one machine group for all machines in the control plane, although you can have multiple machine groups for the workers.
 >
-### hardwareSelector (optional)
+### hardwareSelector (Required)
```
Comment (rahulbabu95) [inline docs/content/en/docs/getting-started/baremetal/bare-spec.md:309]: just to maintain consistency across other sections 
```suggestion
### hardwareSelector (required)
```

File: docs/content/en/docs/getting-started/baremetal/bare-spec.md [hunk 2]
```diff
@@ -365,12 +365,20 @@ When you generate a Bare Metal cluster configuration, the `TinkerbellTemplateCon
...
+```
+
 The following shows three `TinkerbellTemplateConfig` examples that you can add to your cluster configuration file to override the values that EKS Anywhere sets: one for Ubuntu, one for RHEL and one for Bottlerocket.
 Most actions used differ for different operating systems.
 
 >**_NOTE:_** For the `stream-image` action, `DEST_DISK` points to the device representing the entire hard disk (for example, `/dev/sda`).
-For UEFI-enabled images, such as Ubuntu and RHEL 9, write actions use `DEST_DISK` to point to the second partition (for example, `/dev/sda2`), with the first being the EFI partition.
+For all the write actions, `DEST_DISK` must be set to point to the root partition.
+For Ubuntu, write actions use `DEST_DISK` to point to the second partition (for example, `/dev/sda2`), with the first being the EFI partition.
+For RHEL 8 and RHEL 9, `DEST_DISK` is partition 1 (for example, `/dev/sda1`).
```
Comment (rahulbabu95) [inline docs/content/en/docs/getting-started/baremetal/bare-spec.md:379]: should this also be "pointed to" partition 1?

File: test/e2e/SKIPPED_TESTS.yaml [hunk 3]
```diff
@@ -65,11 +65,17 @@ skipped_tests:
 - TestTinkerbellKubernetes126UbuntuExternalEtcdSimpleFlow
 # Skipping skip power action tests - Not going to work because e2e test powers on CP and worker node at the same time and worker node times out early waiting for ipxe
 # Skipping a few redundant tests
+# For RHEL 9 Tinkerbell tests, only enable test case for latest kubernetes version 1.30 due to limited testing resources.
 - TestTinkerbellKubernetes126RedHatSimpleFlow
 - TestTinkerbellKubernetes127RedHatSimpleFlow
 - TestTinkerbellKubernetes128RedHatSimpleFlow
 - TestTinkerbellKubernetes129RedHatSimpleFlow
 - TestTinkerbellKubernetes130RedHatSimpleFlow
+- TestTinkerbellKubernetes125RedHat9SimpleFlow
```
Comment (rahulbabu95) [inline test/e2e/SKIPPED_TESTS.yaml:74]: Don't think we need to add 125 tests here as that K8s version is deprecated!

File: test/e2e/tinkerbell_test.go [hunk 4]
```diff
@@ -1405,6 +1405,72 @@ func TestTinkerbellKubernetes130RedHatSimpleFlow(t *testing.T) {
 	runTinkerbellSimpleFlow(test)
 }
 
+func TestTinkerbellKubernetes125RedHat9SimpleFlow(t *testing.T) {
```
Comment (rahulbabu95) [inline test/e2e/tinkerbell_test.go:1408]: Can remove this test and all associated framework functions for 125.

File: docs/content/en/docs/getting-started/baremetal/bare-spec.md (hunk 1 above)
Comment (2ez4szliu) [reply docs/content/en/docs/getting-started/baremetal/bare-spec.md]: done

File: docs/content/en/docs/getting-started/baremetal/bare-spec.md (hunk 2 above)
Comment (2ez4szliu) [reply docs/content/en/docs/getting-started/baremetal/bare-spec.md]: changed to "points to"

File: cmd/integration_test/build/buildspecs/tinkerbell-test-eks-a-cli.yml [hunk 5]
```diff
@@ -53,6 +53,11 @@ env:
     T_TINKERBELL_IMAGE_REDHAT_1_28: "tinkerbell_ci:image_redhat_1_28"
     T_TINKERBELL_IMAGE_REDHAT_1_29: "tinkerbell_ci:image_redhat_1_29"
     T_TINKERBELL_IMAGE_REDHAT_1_30: "tinkerbell_ci:image_redhat_1_30"
+    T_TINKERBELL_IMAGE_REDHAT_9_1_26: "tinkerbell_ci:image_redhat_9_1_26"
+    T_TINKERBELL_IMAGE_REDHAT_9_1_27: "tinkerbell_ci:image_redhat_9_1_27"
+    T_TINKERBELL_IMAGE_REDHAT_9_1_28: "tinkerbell_ci:image_redhat_9_1_28"
+    T_TINKERBELL_IMAGE_REDHAT_9_1_29: "tinkerbell_ci:image_redhat_9_1_29"
+    T_TINKERBELL_IMAGE_REDHAT_9_1_30: "tinkerbell_ci:image_redhat_9_1_30"
```
Comment (sp1999) [inline cmd/integration_test/build/buildspecs/tinkerbell-test-eks-a-cli.yml:60]: These also needs to be added to the quick-e2e-test buildspec [here](https://github.com/aws/eks-anywhere/blob/main/cmd/integration_test/build/buildspecs/quick-test-eks-a-cli.yml#L167)

**Pull request merged**